lib/tools/system-calls.js

```diff
@@ -225,7 +225,7 @@ systemCallMethods.adbExec = async function (cmd, opts = {}) {
         await this.getDevicesWithRetry();
       }
 
-      if (e.stdout) {
+      if (e.code === 0 && e.stdout) {
```

Comment:
I'd delete the entire condition then. Also my god feeling says it may have something to do with **linkerWarningRe** constant